topics-project/__tests__/signup.test.js

```diff
@@ -0,0 +1,241 @@
+    await userEvent.type(confirmPasswordField, 'test123');
+
+    expect(formData.name).toEqual('John Doe');
+    expect(formData.email).toEqual('jdoe@gmail.com');
+    expect(formData.password).toEqual('test123');
+    expect(formData.confirmPassword).toEqual('test123');
+
+  })
+});
+
+describe('Metrics Form', () => {
+  it('happy path: renders metrics form', () => {
+    const formData = {
+      name: "",
+      email: "",
+      password: "",
+      confirmPassword: "",
+      height: 54,
+      weight: 160,
+      goal: ""
+    };
+
+    const handleInputChange = jest.fn();
+    const handleSubmit = jest.fn();
+    const toggleNext = jest.fn();
+    render(<MetricsForm formData={formData} handleInputChange={handleInputChange} handleSubmit={handleSubmit} toggleNext={toggleNext} />)
+
+    const weightField = screen.queryByPlaceholderText('Enter your weight')
+    expect(weightField).toBeInTheDocument()
+  })
+
+  it('sad path: error in info form when no params sent', () => {
+    expect(() => render(<MetricsForm />)).toThrow("Cannot read properties of undefined");
+  })
+
+  it('happy path: form fields fill out as expected', async () => {
+    const formData = {
+      name: "",
+      email: "",
+      password: "",
+      confirmPassword: "",
+      height: "",
+      weight: "",
+      goal: ""
+    };
+
+    const handleInputChange = (e) => {
+      const { name, value } = e.target;
+      formData[name] = formData[name] + value;
+    };
+    const handleSubmit = jest.fn();
+    const toggleNext = jest.fn();
+    render(<MetricsForm formData={formData} handleInputChange={handleInputChange} handleSubmit={handleSubmit} toggleNext={toggleNext} />)
+
+    const weightField = screen.queryByPlaceholderText('Enter your weight');
+    const maleRadioButton = screen.getByLabelText('Male');
+    const goalSelect = screen.getByLabelText('Goal');
+
+    fireEvent.click(maleRadioButton);
+    fireEvent.change(goalSelect, { target: { value: 'maintenance' } });
+
+    expect(maleRadioButton).toBeTruthy();
+    expect(formData.goal).toEqual('maintenance');
+
+  })
+
+  it('sad path: weightField does not fill out with letters', async () => {
+    const formData = {
+      name: "",
+      email: "",
+      password: "",
+      confirmPassword: "",
+      height: 54,
+      weight: 160,
+      goal: ""
+    };
+
+    const handleInputChange = jest.fn();
+    const handleSubmit = jest.fn();
+    const toggleNext = jest.fn();
+    render(<MetricsForm formData={formData} handleInputChange={handleInputChange} handleSubmit={handleSubmit} toggleNext={toggleNext} />)
+
+    const weightField = screen.queryByPlaceholderText('Enter your weight');
+
+    await userEvent.type(weightField, 'xyz');
+
+    expect(weightField.value).toEqual('');
+  })
+
+});
+
+// Integration tests for signup
+// Test action to create new user
+// Check if new user exists in db using /get-user endpoint
+
+describe('Signup integration tests', () => {
```

Comment:
It's not clear for this test that is actually does what the comment explains.
What would have made this test clearer is if the user data was retrieved from the data store.

For example completing the sign up and then calling the data store in the test to confirm that the use is present.

This test relies on the UI to function correctly to verify the user has been created.

Additionally if this is working as expected then it's creating user in the datastore but not deleting them. 

How would this test run if you run it multiple times?

Reply:
Got it, will be fixed before we push this test to main!